src/WebJobs.Extensions/Extensions/Core/Monitoring/SlidingWindowTraceFilter.cs

```diff
@@ -110,6 +114,11 @@ protected virtual bool FilterCore(TraceEvent traceEvent)
             return passesFilter;
         }
 
+        internal void AddEvent(TraceEvent traceEvent)
+        {
+            _events.Add(traceEvent);
```

Comment:
Need to lock around this as well. Basically around any code that is modifying/enumerating this collection.